### ISSUE TITLE
[Bug]: Unhandled Exception Downloading or Processing Auto-generated Subtitles

### ISSUE THREAD

**Issue by lamusmaser** [tubearchivist/tubearchivist#496]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Unraid, Synology

## Describe the bug
During the `download_pending` task, if a video has either malformed or singular responses to the auto-generated subtitles, then the `_get_auto_caption()` method will throw an exception. This seems to occur for the first few hours after a video is uploaded, while the captions are still generating. This is then causing the queue to die and not move on to either the next video or wrap up the task cleanly.

## Steps To Reproduce
After a video has been uploaded but has not yet generated the auto-generated subtitles, attempting to download the video will cause an exception to be thrown and the task will fail out.

## Expected behavior
The method should handle the exception and return either `False` or the returned subtitles and then move on to the next step in the task.

### Relevant log output

```shell
{"log":"[2023-07-01 02:17:10,811: WARNING/ForkPoolWorker-32] UMO52N2vfk0: get metadata from youtube\n","stream":"stderr","time":"2023-07-01T06:17:10.811269109Z"}
{"log":"[2023-07-01 02:17:15,983: WARNING/ForkPoolWorker-32] UCXuqSBlHAE6Xw-yeJA0Tunw: get metadata from es\n","stream":"stderr","time":"2023-07-01T06:17:15.983303501Z"}
{"log":"[2023-07-01 02:17:16,322: WARNING/ForkPoolWorker-32] UMO52N2vfk0: get sponsorblock timestamps\n","stream":"stderr","time":"2023-07-01T06:17:16.322546337Z"}
{"log":"[2023-07-01 02:17:16,509: WARNING/ForkPoolWorker-32] UMO52N2vfk0: sponsorblock failed: 404\n","stream":"stderr","time":"2023-07-01T06:17:16.509225364Z"}
{"log":"[2023-07-01 02:17:16,509: WARNING/ForkPoolWorker-32] UMO52N2vfk0-en: get user uploaded subtitles\n","stream":"stderr","time":"2023-07-01T06:17:16.509950099Z"}
{"log":"[2023-07-01 02:17:16,509: WARNING/ForkPoolWorker-32] UMO52N2vfk0-en: get auto generated subtitles\n","stream":"stderr","time":"2023-07-01T06:17:16.509988281Z"}
{"log":"[2023-07-01 02:17:16,511: WARNING/ForkPoolWorker-32] 0ae8431b-b1dc-44e8-be84-bc3f1f034dc2 Failed callback\n","stream":"stderr","time":"2023-07-01T06:17:16.511717332Z"}
{"log":"[2023-07-01 02:17:16,512: ERROR/ForkPoolWorker-32] Task download_pending[0ae8431b-b1dc-44e8-be84-bc3f1f034dc2] raised unexpected: IndexError('list index out of range')\n","stream":"stderr","time":"2023-07-01T06:17:16.512903142Z"}
{"log":"Traceback (most recent call last):\n","stream":"stderr","time":"2023-07-01T06:17:16.512911448Z"}
{"log":"  File \"/root/.local/lib/python3.10/site-packages/celery/app/trace.py\", line 451, in trace_task\n","stream":"stderr","time":"2023-07-01T06:17:16.512914714Z"}
{"log":"    R = retval = fun(*args, **kwargs)\n","stream":"stderr","time":"2023-07-01T06:17:16.51291799Z"}
{"log":"  File \"/root/.local/lib/python3.10/site-packages/celery/app/trace.py\", line 734, in __protected_call__\n","stream":"stderr","time":"2023-07-01T06:17:16.512920946Z"}
{"log":"    return self.run(*args, **kwargs)\n","stream":"stderr","time":"2023-07-01T06:17:16.512923971Z"}
{"log":"  File \"/app/home/tasks.py\", line 193, in download_pending\n","stream":"stderr","time":"2023-07-01T06:17:16.512926797Z"}
{"log":"    VideoDownloader(task=self).run_queue(auto_only=auto_only)\n","stream":"stderr","time":"2023-07-01T06:17:16.512929672Z"}
{"log":"  File \"/app/home/src/download/yt_dlp_handler.py\", line 178, in run_queue\n","stream":"stderr","time":"2023-07-01T06:17:16.512932568Z"}
{"log":"    vid_dict = index_new_video(\n","stream":"stderr","time":"2023-07-01T06:17:16.512935463Z"}
{"log":"  File \"/app/home/src/index/video.py\", line 434, in index_new_video\n","stream":"stderr","time":"2023-07-01T06:17:16.512938279Z"}
{"log":"    video.check_subtitles()\n","stream":"stderr","time":"2023-07-01T06:17:16.512955231Z"}
{"log":"  File \"/app/home/src/index/video.py\", line 392, in check_subtitles\n","stream":"stderr","time":"2023-07-01T06:17:16.512958757Z"}
{"log":"    subtitles = handler.get_subtitles()\n","stream":"stderr","time":"2023-07-01T06:17:16.512961673Z"}
{"log":"  File \"/app/home/src/index/subtitle.py\", line 45, in get_subtitles\n","stream":"stderr","time":"2023-07-01T06:17:16.512964548Z"}
{"log":"    auto_cap = self._get_auto_caption(lang)\n","stream":"stderr","time":"2023-07-01T06:17:16.512967484Z"}
{"log":"  File \"/app/home/src/index/subtitle.py\", line 65, in _get_auto_caption\n","stream":"stderr","time":"2023-07-01T06:17:16.512970339Z"}
{"log":"    subtitle = [i for i in all_formats if i[\"ext\"] == \"json3\"][0]\n","stream":"stderr","time":"2023-07-01T06:17:16.512973255Z"}
{"log":"IndexError: list index out of range\n","stream":"stderr","time":"2023-07-01T06:17:16.512976301Z"}
```

### Anything else?

This is related to an issue found during the troubleshooting for https://discord.com/channels/920056098122248193/1128567478621057024. Associated logs are provided from this issue.

**Comment by bbilly1** [comment]
sounds good, we can skip subtitles if they fail at first. they will get grabbed again with the metadata refresh task anyways.

**Comment by bbilly1** [comment]
thanks for reporting, that's now merged in v0.4.0, please update.
